### PR TITLE
Improve code blocks and add support for TOC

### PR DESCRIPTION
**Pull request by patrick91**
This PR includes a custom rehype plugin for handling code blocks and TOCs.

Code blocks now support:

### Highlighting words individually

``````
```python highlight=strawberry,str
import strawberry

@strawberry.type
class X:
    name: str
```
``````

### Highlighting lines 

``````
```python line=1-4
import strawberry

@strawberry.type
class X:
    name: str
```
``````

### Add notes to code comments

This is probably not implemented in the best way, but for now it works:

``````
```python
import ^[info](strawberry)

@strawberry.type
class X:
    name: str
```

<CodeNotes id="info">Strawberry is a cool library</CodeNotes>
``````

It also adds TOC support, by using `toc: true` in the frontmatter section:)

You can see this in action here: https://strawberryrocks-oyzofz3nu-strawberry-graphql.vercel.app/docs/tag/84316c1ab57a5116b8f60b93f9f4ff214893570f/test

### CODE REVIEW

**Comment by jkimbo** [comment]
The highlighting is pretty subtle. Can we make it more obvious?

CleanShot 2021-10-04 at 13 50 44@2x

File: lib/api.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ const octokit = new customOctokit({
 export const OWNER = "strawberry-graphql";
 export const REPO = "strawberry";
 // default branch to query against.
-export const REF = "main";
+export const REF = "docs-improvements";
```
Comment (jkimbo) [inline lib/api.ts:29]: ‼️ Did you mean to commit this?

Comment (patrick91) [reply lib/api.ts:29]: nope! #fail

**Comment by patrick91** [comment]
> The highlighting is pretty subtle. Can we make it more obvious?
> 
> CleanShot 2021-10-04 at 13 50 44@2x

Yes, wanted to do that in another PR :)

File: lib/api.ts (hunk 1 above)
Comment (patrick91) [reply lib/api.ts:29]: fixed on main! thanks for checking 😊

**Pull request merged**